contracts/protocol/bases/SellerBase.sol

```diff
@@ -39,6 +41,34 @@ contract SellerBase is ProtocolBase, IBosonAccountEvents {
         // Cache protocol lookups for reference
         ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
 
+        // Get message sender
+        address sender = msgSender();
+
+        // Do caller and uniqueness checks based on auth type
+        if (_seller.admin == address(0)) {
+            // Check that caller owns the auth token
+            address authTokenContract = lookups.authTokenContracts[_authToken.tokenType];
+            address tokenIdOwner = IERC721(authTokenContract).ownerOf(_authToken.tokenId);
+            require(tokenIdOwner == sender, NOT_ADMIN);
+
+            // Check that auth token is unique to this seller
+            require(
+                lookups.sellerIdByAuthToken[_authToken.tokenType][_authToken.tokenId] == 0,
+                AUTH_TOKEN_MUST_BE_UNIQUE
+            );
+        } else {
+            // Check that caller is supplied admin
+            require(_seller.admin == sender, NOT_ADMIN);
+
+            // Check that the admin address is unique to one seller id, across all roles
+            require(
+                lookups.sellerIdByOperator[_seller.admin] == 0 &&
+                    lookups.sellerIdByAdmin[_seller.admin] == 0 &&
+                    lookups.sellerIdByClerk[_seller.admin] == 0,
+                SELLER_ADDRESS_MUST_BE_UNIQUE
+            );
+        }
+
         // Check active is not set to false
         require(_seller.active, MUST_BE_ACTIVE);
```

Comment:
is it not worth doing this first ? Cause if this is false we can skip all of the above ?